File: 14.Longest_Common_Prefix.cpp
class Solution {
    public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return {};
        string prefix = strs[0];
        for (int i=1; i<strs.size(); i++){
            prefix = helper(prefix, strs[i]);
        }
        return prefix;
    }
    
    string helper(string prefix, string str) {
        string res;
        int index = 0;
        while (prefix[index] == str[index] && index<prefix.length() && index<str.length()){
            res += prefix[index];
            index++;
        }
        return res;
    }
};

File: 14.LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        prefix = strs[0]
        for s in strs:
            prefix = self.check_prefix(prefix, s)
        return prefix
    
    def check_prefix(self, prefix, string):
        length = 0
        while length < len(prefix) and length < len(string):
            if prefix[length] == string[length]:
                length += 1
            else:
                break
        return prefix[0:length]